ingress/controller.go

```diff
@@ -104,3 +113,24 @@ func (c *controller) Stop() {
 	log.Info("Stopping controller")
 	close(c.stop)
 }
+
+func (c *controller) Healthy() bool {
+	return c.started.get()
+}
+
+type safeBool struct {
+	lock sync.Mutex
+	val  bool
```

Comment:
lol so go doesn't have a concept of a volatile variable :-/

I started reading this and it told me to stop trying to be clever and use a mutex :)